keps/sig-node/2400-node-swap/README.md

```diff
@@ -166,6 +170,71 @@ administrators can configure the kubelet such that:
+
+Allocate the swap limit equal to the requested memory for each container and adjust the proportion of swap based on the total swap memory available, we can follow these steps:
+
+1. **Calculate the total memory requests of the pod:**
+   - Sum up the memory requests of all containers in the pod. Let's call this value `TotalMemory`.
+
+2. **Determine the swap proportion for each container:**
+   - For each container, divide its memory request by the `TotalMemory`. The result will be the proportion of requested memory for that container within the pod. Let's call this value `RequestedMemoryProportion`.
+
+3. **Calculate the total swap memory available:**
+   - Determine the total swap memory available in the system. If the total swap memory is equal to the total physical memory, you can skip this step. Otherwise, calculate the proportion of available swap memory compared to the total physical memory. Let's call this value `SwapMemoryProportion`.
+
+4. **Calculate the swap limit for each container:**
+   - Multiply the `RequestedMemoryProportion` of a container by its memory request and then multiply the result by the `SwapMemoryProportion`. The result will be the adjusted swap limit for that specific container within the pod.
+
+#### Example 1 - Configured swap memory on the host is equal to the physical memory
+Suppose we have a Burstable QoS pod with two containers:
+
+- Container A: Memory request 2 GB
+- Container B: Memory request 1 GB
+
+Let's assume the total physical memory is 4 GB and the total swap memory available is also 4 GB.
+
+Step 1: Calculate the total memory requests of the pod: 2 GB + 1 GB = 3 GB
+
+Step 2: Determine the requested memory proportion for each container:
+- Container A: (2 GB) / (3 GB) = 2/3
+- Container B: (1 GB) / (3 GB) = 1/3
+
+Step 3: Calculate the total swap memory available: Since the total swap memory (4 GB) is equal to the total physical memory (4 GB), the `SwapMemoryProportion` will be 1.
+
+Step 4: Calculate the swap limit for each container:
+- Container A: (2/3) * 2 GB * 1 = 4/3 GB ≈ 1.33 GB
+- Container B: (1/3) * 1 GB * 1 = 1/3 GB ≈ 0.33 GB
+
+In this example, Container A would have a swap limit of 1.33 GB, and Container B would have a swap limit of 0.33 GB.
+
+This approach allocates swap limits based on each container's memory request and adjusts the proportion based on the total swap memory available in the system. It ensures that each container gets a fair share of the swap space and helps maintain resource allocation efficiency.
+
+#### Example 2 - Configured swap memory on the host is NOT equal to the physical memory
+
+If the total physical memory is 4 GB and the total swap memory available is 2 GB, we can adjust the example 1 using the same approach:
```

Comment:
nit:
What do you think about making this example a bit more realistic?
Maybe we can show total memory of `400G` and swap size of `30G`?

Not sure it's important though.